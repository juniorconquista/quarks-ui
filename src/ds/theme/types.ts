import {
    ColorProps,
    FlexboxProps,
    LayoutProps,
    SpaceProps,
    PositionProps,
    BorderProps,
    GridProps,
} from 'styled-system';

export type DefaultStyleProps = BorderProps &
    ColorProps &
    FlexboxProps &
    LayoutProps &
    PositionProps &
    SpaceProps &
    GridProps & {
        // for some reason this isn't included; this is for replacing the 'base' component
        // as?: React.ComponentType;
    };
