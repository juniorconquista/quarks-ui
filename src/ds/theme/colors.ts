export default {
    dark: false,
    primary: {
        default: '#F26522',
        dark: '#d14b0c',
        light: '#f69263',
        contrast: '#ffffff',
    },
    secondary: {
        default: '#372E2A',
        dark: '#2c2522',
        light: '#453935',
        contrast: '#1B2B3A',
    },
    danger: {
        default: '#ff6961',
        dark: '#ff261b',
        light: '#ffbdb9',
        contrast: '#ffffff',
    },
    success: {
        default: '#0E9d51',
        dark: '#0a753c',
        light: '#10bc60',
        contrast: '#ffffff',
    },
    warning: {
        default: '#ffb143',
        dark: '#e4902c',
        light: '#ffd868',
        contrast: '#ffffff',
    },
    info: {
        default: '#0ea7a7',
        dark: '#0a7676',
        light: '#11d4d4',
        contrast: '#ffffff',
    },
    grays: {
        default: '#9ba8b2',
        dark: '#4b4b4b',
        light: '#eeeeef',
        contrast: '#f5f5f5',
    },
    background: {
        default: '#ffffff',
        dark: '#F26522',
    },
    text: {
        light: '#ffffff',
        dark: '#1B2b3a',
    },
    disabled: '#9398a6',
    primaries: ['#d7edf4'],
    whites: ['#eeeeef', '#f2fbfe'],
    blacks: ['#35363a'],
    hovers: ['#f5f7f7'],
};
