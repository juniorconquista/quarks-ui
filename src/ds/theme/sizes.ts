export const breakpoints = ['0rem', '37.5rem', '62rem'];

export const space = {
    none: '0',
    xxxs: '0.285rem',
    xxs: '0.571rem',
    xs: '0.857rem',
    sm: '1.142rem',
    md: '1.714rem',
    lg: '2.5rem',
    xl: '3.428rem',
    xxl: '6.857rem',
};

export const fontSizes = [
    '0.5rem',
    '0.85rem',
    '1rem',
    '1.5rem',
    '1.8rem',
    '2rem',
];

export const buttonSizes = {
    medium: {
        paddingTop: 8,
        paddingBottom: 8,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
    small: {
        paddingTop: 4,
        paddingBottom: 4,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
};

export const radii = {
    none: '0',
    sm: '0.285rem',
    md: '2.25rem',
    lg: '1.142rem',
    xl: '1.714rem',
};
