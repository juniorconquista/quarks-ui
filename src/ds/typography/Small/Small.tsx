import styled from 'styled-components';
import { variant, VariantArgs, color, typography, space } from 'styled-system';
import { SmallStyledProps } from './small.types';

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            color: 'text',
            fontSize: 'fontSizes.2',
        },
    },
} as VariantArgs);

export const Small: React.FC<SmallStyledProps> = styled.small`
    ${variants}
    ${color}
    ${typography}
    ${space}
`;

export default Small;
