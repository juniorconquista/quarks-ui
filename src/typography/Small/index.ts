export { default } from './Small'
export * from './small.types'
