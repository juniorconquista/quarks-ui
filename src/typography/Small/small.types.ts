import { DefaultStyleProps } from '../../theme/types';
import { TypographyProps } from 'styled-system';

export type SmallStyledProps = DefaultStyleProps & TypographyProps;

export interface SmallProps extends SmallStyledProps {
    /** Each styling variant has unique properties that are inherited through the type informed */
    styling?: 'base';
}
