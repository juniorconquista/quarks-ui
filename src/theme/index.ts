import colors from './colors';
import { breakpoints, space, fontSizes, radii } from './sizes';

const theme = {
    fontSizes,
    space,
    breakpoints,
    radii,
    colors: colors(),
};

//@ts-ignore
theme.breakpoints.mobile = breakpoints[0];
//@ts-ignore
theme.breakpoints.tablet = breakpoints[1];
//@ts-ignore
theme.breakpoints.desktop = breakpoints[2];

export default theme;
