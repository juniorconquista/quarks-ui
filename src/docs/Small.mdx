---
name: Small
menu: Typography
---

import { ThemeProvider } from 'styled-components';
import { Playground, Props } from 'docz';
import { Small } from '../ds/typography';
import * as theme from '../ds/theme';

# Small

Search engines use the headings to index the structure and content of your web pages. Users often skim a page by its headings. It is important to use headings to show the document structure.

Each HTML heading has a default size. However, you can specify the size for any heading with the style attribute, using the CSS _font-size_ property.

### Best practices

-   Use HTML headings for headings only. Don't use headings to make text BIG or bold.

## Properties

<Props of={Small} />

## Basic usage

<Playground>
    <ThemeProvider theme={{ ...theme.default }}>
        <Small styling="base">Base</Small>
    </ThemeProvider>
</Playground>

## Customization

### Color

The color utility parses a component's _color_ and _bg_ props and converts them into CSS declarations. By default the raw value of the prop is returned. Color palettes can be configured with the ThemeProvider to use keys as prop values, with support for dot notation.

You can see all the color customization [here](https://styled-system.com/api/#color).

```javascript
// picks the value defined in `theme.colors.blue`
<Small styling="base" color='blue'>Title</Small>

// picks up a nested color value using dot notation `theme.colors.gray[0]`
<Small color='gray.0'>Title</Small>

// raw CSS color value
<Small color='#f00'>Title</Small>

// background colors
<Small bg='blue'>Title</Small>

// verbose prop
<Small backgroundColor='blue'>Title</Small>
```

### Space

The space utility converts shorthand margin and padding props to _margin_ and _padding_ CSS declarations.

You can see all the space customization [here](https://styled-system.com/api#space).

```javascript
// sets margin value of `theme.space[2]`
<Small styling="base" m={2}>Title</Small>

// sets a margin value of `16px` since it's greater than `theme.space.length`
<Small m={16}>Title</Small>

// sets margin `'auto'`
<Small m='auto'>Title</Small>

// sets margin `8px` fom mobile devices, `16px` for tablet and `32px` for desktop from the first breakpoint and up
<Small m={{ mobile:2, tablet:3, desktop:4 }}>Title</Small>
```

### Typography

Typography props include _fontFamily_, _fontSize_, _fontWeight_, _lineHeight_, _letterSpacing_, _textAlign_, and _fontStyle_.

You can see all the typography customization [here](https://styled-system.com/api/#typography).

```javascript
// font-size of `theme.fontSizes[3]`
<Small styling="base" fontSize={3}>Title</Small>

// font-size `32px`
<Small fontSize={32}>Title</Small>

// font-size `'2em'`
<Small fontSize='2em'>Title</Small>

// font-size `10px` on all viewports and `12px` from the first breakpoint and up
<Small fontSize={{ mobile:10, desktop:12 }}>Title</Small>

// fontFamily
<Small fontFamily='mono'>Title</Small>

// textAlign
<Small textAlign='center'>Title</Small>
<Small textAlign={[ 'center', 'left' ]}>Title</Small>

// lineHeight
<Small lineHeight='1.25'>Title</Small>

// fontWeight
<Small fontWeight='bold'>Title</Small>

// letterSpacing
<Small letterSpacing='0.1em'>Title</Small>
```
