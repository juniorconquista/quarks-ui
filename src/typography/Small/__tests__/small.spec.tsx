import React from 'react';
import renderer from 'react-test-renderer';
import ThemeTest from '../../../utils/ThemeProviderTest';
import 'jest-styled-components';

import Small from '../Small';
import theme from '../../../theme';

describe('Small', () => {
    test('basic snapshot', () => {
        const small = renderer
            .create(
                <ThemeTest>
                    <Small styling="base" />
                </ThemeTest>,
            )
            .toJSON();
        expect(small).toMatchSnapshot();
    });

    test('should inherit the styling properties of the base variant', () => {
        const small = renderer
            .create(
                <ThemeTest>
                    <Small styling="base" />
                </ThemeTest>,
            )
            .toJSON();
        expect(small).toHaveStyleRule('color', theme.colors.text.default);
        expect(small).toHaveStyleRule('font-size', theme.fontSizes[2]);
    });

    test('should apply custom styling properties', () => {
        const small = renderer
            .create(<Small styling="base" mt="20px" p="3px" />)
            .toJSON();
        expect(small).toHaveStyleRule('margin-top', '20px');
        expect(small).toHaveStyleRule('padding', '3px');
    });
});
