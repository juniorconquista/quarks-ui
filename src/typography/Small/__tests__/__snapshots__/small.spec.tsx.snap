// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small basic snapshot 1`] = `
.c0 {
  color: #343a40;
  font-size: 0.83rem;
}

<small
  className="c0"
/>
`;
